providers/providers.go

```diff
@@ -31,7 +31,7 @@ type ProviderManager struct {
 	// all non channel fields are meant to be accessed only within
 	// the run method
 	providers *lru.LRU
-	dstore    ds.Datastore
+	dstore    *autobatch.Datastore
```

Comment:
are we concerned about the lack of thread-safety in the autobatching datastore?

Reply:
We're only using it from one thread here.